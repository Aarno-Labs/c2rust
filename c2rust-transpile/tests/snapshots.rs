use std::fs;

#[test]
fn transpile() {
    let config = || c2rust_transpile::TranspilerConfig {
        dump_untyped_context: false,
        dump_typed_context: false,
        pretty_typed_context: false,
        dump_function_cfgs: false,
        json_function_cfgs: false,
        dump_cfg_liveness: false,
        dump_structures: false,
        verbose: false,
        debug_ast_exporter: false,
        incremental_relooper: true,
        fail_on_multiple: false,
        filter: None,
        debug_relooper_labels: false,
        prefix_function_names: None,
        translate_asm: true,
        use_c_loop_info: true,
        use_c_multiple_info: true,
        simplify_structures: true,
        panic_on_translator_failure: false,
        emit_modules: false,
        fail_on_error: false,
        replace_unsupported_decls: c2rust_transpile::ReplaceMode::Extern,
        translate_valist: true,
        overwrite_existing: true,
        reduce_type_annotations: false,
        reorganize_definitions: false,
        enabled_warnings: Default::default(),
        emit_no_std: false,
        output_dir: None,
        translate_const_macros: false,
        translate_fn_macros: false,
        disable_refactoring: false,
        preserve_unused_functions: false,
        log_level: log::LevelFilter::Warn,
        emit_build_files: false,
        binaries: Vec::new(),
    };
    insta::glob!("snapshots/*.c", |f: &Path| {
        let (_temp_dir, temp_path) =
            c2rust_transpile::create_temp_compile_commands(&[f.to_owned()]);
        c2rust_transpile::transpile(config(), &temp_path, &[]);
        let output_path = f.with_extension("rs");
        let output = fs::read_to_string(&output_path).unwrap();
        fs::remove_file(&output_path).unwrap();
        insta::assert_snapshot!(&output);
    });
}
