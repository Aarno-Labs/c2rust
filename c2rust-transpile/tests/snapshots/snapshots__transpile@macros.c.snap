---
source: c2rust-transpile/tests/snapshots.rs
expression: cat tests/snapshots/macros.rs
input_file: c2rust-transpile/tests/snapshots/macros.c
---
#![allow(
    dead_code,
    mutable_transmutes,
    non_camel_case_types,
    non_snake_case,
    non_upper_case_globals,
    unused_assignments,
    unused_mut
)]
extern "C" {
    fn extern_fn() -> std::ffi::c_int;
}
pub type U64 = std::ffi::c_ulonglong;
#[derive(Copy, Clone)]
#[repr(C)]
pub struct fn_ptrs {
    pub v: *mut std::ffi::c_void,
    pub fn1: Option<unsafe extern "C" fn() -> std::ffi::c_int>,
    pub fn2: Option<unsafe extern "C" fn(std::ffi::c_int) -> std::ffi::c_int>,
}
pub type zstd_platform_dependent_type = std::ffi::c_long;
#[no_mangle]
pub unsafe extern "C" fn test_fn_macro(mut x: std::ffi::c_int) -> std::ffi::c_int {
    return x * x;
}
#[no_mangle]
pub unsafe extern "C" fn reference_define() -> std::ffi::c_int {
    let mut x: std::ffi::c_int = 1 as std::ffi::c_int;
    x += 2 as std::ffi::c_int;
    if (3 as std::ffi::c_int) < (2 as std::ffi::c_int + 1 as std::ffi::c_int) * 3 as std::ffi::c_int
    {
        x += (2 as std::ffi::c_int + 1 as std::ffi::c_int) * 3 as std::ffi::c_int;
    }
    return x;
}
#[no_mangle]
pub static mut fns: fn_ptrs = {
    let mut init = fn_ptrs {
        v: 0 as *const std::ffi::c_void as *mut std::ffi::c_void,
        fn1: None,
        fn2: None,
    };
    init
};
#[no_mangle]
pub static mut p: *const fn_ptrs = unsafe { &fns as *const fn_ptrs };
#[no_mangle]
pub unsafe extern "C" fn test_zstd() -> U64 {
    return (if ::core::mem::size_of::<zstd_platform_dependent_type>() as std::ffi::c_ulong
        == 4 as std::ffi::c_ulong
    {
        30 as std::ffi::c_int
    } else {
        31 as std::ffi::c_int
    }) as U64;
}
#[no_mangle]
pub unsafe extern "C" fn stmt_expr_inc() -> std::ffi::c_int {
    let mut a: std::ffi::c_int = 0 as std::ffi::c_int;
    let mut b: *mut std::ffi::c_int = &mut a;
    ({
        *b += 1;
        *b;
        *b;
        *b
    });
    return ({
        *b += 1;
        *b;
        *b
    });
}
#[no_mangle]
pub unsafe extern "C" fn test_switch(mut x: std::ffi::c_int) -> std::ffi::c_int {
    match x {
        1 => return 10 as std::ffi::c_int,
        2 => return 20 as std::ffi::c_int,
        _ => {}
    }
    return 0 as std::ffi::c_int;
}
#[no_mangle]
pub unsafe extern "C" fn test_silk_int16_MIN() -> std::ffi::c_int {
    let mut _null: std::ffi::c_char =
        (*::core::mem::transmute::<&[u8; 1], &[std::ffi::c_char; 1]>(b"\0"))[(0x8000
            as std::ffi::c_int
            as std::ffi::c_short
            as std::ffi::c_int
            + 0x8000 as std::ffi::c_int)
            as usize];
    return 0x8000 as std::ffi::c_int as std::ffi::c_short as std::ffi::c_int;
}
#[no_mangle]
pub unsafe extern "C" fn use_extern_value() -> std::ffi::c_int {
    return extern_fn();
}
#[no_mangle]
pub unsafe extern "C" fn local_fn() -> std::ffi::c_int {
    return 1234 as std::ffi::c_int;
}
#[no_mangle]
pub unsafe extern "C" fn use_local_value() -> std::ffi::c_int {
    return local_fn();
}
